lib/mana/engines/javascript.rb

```
@@ -21,38 +66,210 @@ def self.create_context
+          attached << "ruby.write"
+        end
+
+        # Remote reference operations
+        attach_ref_callbacks(ctx, attached)
+
+        # Attach methods from the caller's receiver
+        attach_receiver_methods(ctx, attached)
+
+        # Attach registered Mana effects
+        attach_effects(ctx, attached)
+      end
```

RuboCop (1.84.2)

[convention] Assignment Branch Condition size for `attach_ruby_callbacks` is too high. [<10, 24, 6> 26.68/23]

(Metrics/AbcSize)